### ISSUE TITLE
pegomock generation is not idempotent

### ISSUE THREAD

**Issue by hferentschik** [jenkins-x/jx#3502]
### Summary 

The output of `make generate-mocks` changes when run multiple times even without any changes to the code.

### Steps to reproduce the behavior

Without changing any code run:

`make generate-mocks`

### Expected behavior

Running pegomock code generation when no code changes have been made should not change any mocks.

### Actual behavior

There seems to be a couple of things wrong. First, we have several interfaces.go for which peogomock generates mocks in the form `versioned[0-9]_interface.go`. However, the order in which these files get generated changes on each run so that even the overall outcome is the same, we get modified files in terms of version control.

```
modified:   pkg/jx/cmd/clients/mocks/matchers/versioned0_interface.go
modified:   pkg/jx/cmd/clients/mocks/matchers/versioned1_interface.go
modified:   pkg/jx/cmd/clients/mocks/matchers/versioned3_interface.go
modified:   pkg/jx/cmd/clients/mocks/matchers/versioned4_interface.go
modified:   pkg/jx/cmd/clients/mocks/matchers/versioned_interface.go
```

The other issue is that the import ordering of pegomock does not seem to in sync with `go fmt`. Running `make generate-mocks` creates changes like:

```
 import (
+       "reflect"
        "github.com/petergtz/pegomock"
        terminal "gopkg.in/AlecAivazis/survey.v1/terminal"
-       "reflect"
 )
```

which will then be reverted again by `make fmt`.

This relates to issue #3501

**Comment by hferentschik** [comment]
Without a fix in Pegomock we cannot get rid of all the problems since at the very least the `MockFactory` keeps changing. See also https://github.com/petergtz/pegomock/issues/82.

For now, I would like to do the following:

* Clean up the Makefile and add a recipe to fail the build in case `make generate` does modify anything except the `MockFactory`
* Disable generation of matchers. Most are not used and they also create files which change on each run. If needed one can easily create a matcher manually. Also once the Pegomock issue is fixed we could generate them again
* For simplifying this cleanup and because of some of the heavily mocked test not carrying their weight, I'd like to delete some tests

**Comment by hferentschik** [comment]
merged via pull request #3545
